plugins/net_plugin/include/eosio/net_plugin/net_utils.hpp

```diff
@@ -40,29 +40,71 @@ namespace detail {
+         return {};
+
+      }
+      string::size_type colon = peer_add.find(':', end_bracket+1);
+      if (colon == string::npos || colon == 0) {
+         return {};
+      }
+      string::size_type colon2 = peer_add.find(':', colon + 1);
+      string host = (end_bracket > 0) ? peer_add.substr( 0, end_bracket+1 ) : peer_add.substr( 0, colon );
+      string port = peer_add.substr( colon + 1, colon2 == string::npos ? string::npos : colon2 - (colon + 1));
+      string remainder = colon2 == string::npos ? "" : peer_add.substr( colon2 + 1 );
+      return {std::move(host), std::move(port), std::move(remainder)};
+   }
+
 } // namespace detail
 
-   /// @return listen address and block sync rate limit (in bytes/sec) of address string
+   /// @return host, port, type. returns empty on invalid peer_add, does not throw
+   inline std::tuple<std::string, std::string, std::string> split_host_port_type(const std::string& peer_add) {
+
+      using std::string;
+      // host:port[:trx|:blk][:<rate>]   // rate is discarded
+      if (peer_add.empty()) return {};
+
+      constexpr bool should_throw = false;
+      auto [host, port, remainder] = detail::split_host_port_remainder(peer_add, should_throw);
+      if (host.empty()) return {};
+
+      string::size_type end = remainder.find_first_of( " :+=.,<>!$%^&(*)|-#@\t" ); // future proof by including most symbols without using regex
+      std::string type = remainder.substr(0, end);
+
+      return {std::move(host), std::move(port), std::move(type)};
+   }
+
+   /// @return listen address, type [trx|blk], and block sync rate limit (in bytes/sec) of address string
+   /// @throws chain::plugin_config_exception on invalid address
    inline std::tuple<std::string, size_t> parse_listen_address( const std::string& address ) {
-      auto listen_addr = address;
-      auto limit = std::string("0");
-      auto last_colon_location = address.rfind(':');
-      if( auto right_bracket_location = address.find(']'); right_bracket_location != address.npos ) {
-         if( std::count(address.begin()+right_bracket_location, address.end(), ':') > 1 ) {
-            listen_addr = std::string(address, 0, last_colon_location);
-            limit = std::string(address, last_colon_location+1);
-         }
-      } else {
-         if( auto colon_count = std::count(address.begin(), address.end(), ':'); colon_count > 1 ) {
-            EOS_ASSERT( colon_count <= 2, chain::plugin_config_exception,
-                        "Invalid address specification ${addr}; IPv6 addresses must be enclosed in square brackets.", ("addr", address));
-            listen_addr = std::string(address, 0, last_colon_location);
-            limit = std::string(address, last_colon_location+1);
-         }
+      constexpr bool should_throw = true;
+      auto [host, port, remainder] = detail::split_host_port_remainder(address, should_throw);
+      auto listen_addr = host + ":" + port;
```

Comment:
Is ":" a valid listen address?
For example, when `peer_add = "192.168.0.1"`, I think `split_host_port_remainder` would return 3 empty strings without throwing, which would cause `listen_addr ` to be ":".

My suggestion at line 63 would address this I think.

Reply:
https://github.com/AntelopeIO/spring/pull/1115/commits/6c2442b14f7724c44625cc8b8f862b1cada1449d